### PR TITLE
Replace deprecated functions in Auth plugin

### PR DESCRIPTION
**Pull request by qifeng-bai**

### CODE REVIEW

File: grails-app/services/au/org/ala/specieslist/UserDetailsService.groovy [hunk 1]
```diff
@@ -37,123 +37,59 @@ class UserDetailsService {
...
+                        log.error "No user or userId found for username: ${list.username} || user = ${user}"
+                        log.error "Setting username to info@ala.org.au"
+                        list.username = "info@ala.org.au"
+                        list.userId = 2729
+                        result[DEFAULT] += 1
+                    } else {
+                        log.error "No user or userId found for username: ${list.username} || user = ${user}"
+                        log.error "Setting username to 0"
+                        list.userId = 0
+                        result[INVALID] += 1
```
Comment (adam-collins) [inline grails-app/services/au/org/ala/specieslist/UserDetailsService.groovy:73]: `list.save(flush: true)` is missing from the refactor

File: grails-app/controllers/au/org/ala/specieslist/AdminController.groovy [hunk 2]
```diff
@@ -79,7 +81,12 @@ class AdminController {
...
     // Access managed by AdminInterceptor
     def updateListsWithUserIds() {
-        Boolean successful = userDetailsService.updateSpeciesListUserDetails()
-        render(status: successful ? 200 : 404, text: "Update of editors was ${(successful)?'':'not'} successful")
+        def result = userDetailsService.updateSpeciesListUserDetails()
+        render result as JSON
+    }
+
+    def updateListsEditorWithIds() {
+        def result = userDetailsService.updateEditorsList()
```
Comment (adam-collins) [inline grails-app/controllers/au/org/ala/specieslist/AdminController.groovy:89]: `userDetailsService.updateEditorsList()` does not exist

**Review by adam-collins — CHANGES_REQUESTED** [review]
I would agree with removing `updateSpeciesListUserDetails`, consistent with removing `updateEditorsList`, because all it appears to do is update the userId for the userId/email list owner fields and the reason to do that is no longer valid.

File: grails-app/controllers/au/org/ala/specieslist/AdminController.groovy (hunk 2 above)
Comment (qifeng-bai) [reply grails-app/controllers/au/org/ala/specieslist/AdminController.groovy:89]: The method updateListsEditorWithIds() in AdminController is intended to be removed.
Removed

File: grails-app/services/au/org/ala/specieslist/UserDetailsService.groovy (hunk 1 above)
Comment (qifeng-bai) [reply grails-app/services/au/org/ala/specieslist/UserDetailsService.groovy:73]: The Git PR page does not display the changes properly.

list.save() was moved to https://github.com/AtlasOfLivingAustralia/specieslist-webapp/blob/develop/grails-app/services/au/org/ala/specieslist/UserDetailsService.groovy#L145

File: grails-app/controllers/au/org/ala/specieslist/AdminController.groovy (hunk 2 above)
Comment (qifeng-bai) [reply grails-app/controllers/au/org/ala/specieslist/AdminController.groovy]: updateListsEditorWithIds() in AdminController is meant to be removed

**Comment by qifeng-bai** [comment]
> I would agree with removing `updateSpeciesListUserDetails`, consistent with removing `updateEditorsList`, because all it appears to do is update the userId for the userId/email list owner fields and the reason to do that is no longer valid.

I currently plan to retain this function. In contrast, I removed the outdated editor update function because the database structure for the editor has been modified, leave the editor update function obsolete.

**Pull request merged**